`make fuzz` target is broken!

Hello.
I was working on the fuzzer when I ran into several problems:
1. `includes` addresses is broken:
https://github.com/davmac314/dinit/blob/ececd169fbfac35035d0ddb6d7cc1d09ae1e5acf/src/tests/cptests/Makefile#L41-L48
(fixed by #92).

2. `prepare-incdir` target not runned by make when building `fuzz` target! its target prepare include dirs for building tests/fuzz (fixed by #94)
3. because `fuzz` target not runned by upstream `MakeFile`, `mconfig.h` not exist but we need `mconfig.h` for buliding `fuzz` target (fixed by #95):
````
❯ make fuzz
clang -O1 -fsanitize=address,undefined,fuzzer-no-link,leak -MMD -MP -Iincludes -I../../../dasynq/include -c ../test-bpsys.cc -o fuzz-test-bpsys.o
clang -O1 -fsanitize=address,undefined,fuzzer-no-link,leak -MMD -MP -Iincludes -I../../../dasynq/include -c ../test-dinit.cc -o fuzz-test-dinit.o
clang -O1 -fsanitize=address,undefined,fuzzer-no-link,leak -MMD -MP -Iincludes -I../../../dasynq/include -c ../../control.cc -o fuzz-control.o
In file included from ../../control.cc:6:
In file included from includes/control.h:16:
In file included from includes/service-listener.h:4:
includes/service-constants.h:4:10: fatal error: 'mconfig.h' file not found
#include "mconfig.h"
         ^~~~~~~~~~~
1 error generated.
````

regards

--- a/src/tests/cptests/Makefile
+++ b/src/tests/cptests/Makefile
@@ -48,7 +48,7 @@ fuzz_objects = $(foreach obj,$(parent_objs),fuzz-$(obj))
 include/mconfig.h: ../../../build/includes/mconfig.h
 	/usr/bin/env mv ../../../build/includes/mconfig.h includes/mconfig.h
 
-fuzz: prepare-incdir gen-mconfig fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
+fuzz: prepare-incdir include/mconfig.h fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
 	clang++ -std=c++11 -g -O1 -Iincludes -I../../../dasynq/include -fsanitize=fuzzer,address,undefined,leak fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects) -o fuzz
 
 $(fuzz_parent_test_objects): fuzz-%.o: ../%.cc

--- a/src/tests/cptests/Makefile
+++ b/src/tests/cptests/Makefile
@@ -45,7 +45,7 @@ fuzz_objects = $(foreach obj,$(parent_objs),fuzz-$(obj))
 ../../../build/includes/mconfig.h: ../../../mconfig
 	$(MAKE) -C ../../../build all
 
-fuzz: prepare-incdir ../../../build/include/mconfig.h fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
+fuzz: prepare-incdir ../../../build/includes/mconfig.h fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
 	clang++ -std=c++11 -g -O1 -Iincludes -I../../../dasynq/include -I../../../build/includes/ -fsanitize=fuzzer,address,undefined,leak fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects) -o fuzz
 
 $(fuzz_parent_test_objects): fuzz-%.o: ../%.cc

--- a/src/tests/cptests/Makefile
+++ b/src/tests/cptests/Makefile
@@ -45,17 +45,14 @@ fuzz_objects = $(foreach obj,$(parent_objs),fuzz-$(obj))
 ../../../build/includes/mconfig.h: ../../../mconfig
 	$(MAKE) -C ../../../build all
 
-include/mconfig.h: ../../../build/includes/mconfig.h
-	/usr/bin/env mv ../../../build/includes/mconfig.h includes/mconfig.h
-
-fuzz: prepare-incdir include/mconfig.h fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
-	clang++ -std=c++11 -g -O1 -Iincludes -I../../../dasynq/include -fsanitize=fuzzer,address,undefined,leak fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects) -o fuzz
+fuzz: prepare-incdir ../../../build/include/mconfig.h fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
+	clang++ -std=c++11 -g -O1 -Iincludes -I../../../dasynq/include -I../../../build/includes/ -fsanitize=fuzzer,address,undefined,leak fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects) -o fuzz
 
 $(fuzz_parent_test_objects): fuzz-%.o: ../%.cc
-	clang -O1 -fsanitize=address,undefined,fuzzer-no-link,leak -MMD -MP -Iincludes -I../../../dasynq/include -c $< -o $@
+	clang -O1 -fsanitize=address,undefined,fuzzer-no-link,leak -MMD -MP -Iincludes -I../../../dasynq/include -I../../../build/includes/ -c $< -o $@
 
 $(fuzz_objects): fuzz-%.o: ../../%.cc
-	clang -O1 -fsanitize=address,undefined,fuzzer-no-link,leak -MMD -MP -Iincludes -I../../../dasynq/include -c $< -o $@
+	clang -O1 -fsanitize=address,undefined,fuzzer-no-link,leak -MMD -MP -Iincludes -I../../../dasynq/include -I../../../build/includes/ -c $< -o $@
 
 
 -include $(objects:.o=.d)

--- a/src/tests/cptests/Makefile
+++ b/src/tests/cptests/Makefile
@@ -39,10 +39,14 @@ fuzz_parent_test_objects = $(foreach obj,$(notdir $(parent_test_objects)),fuzz-$
 fuzz_objects = $(foreach obj,$(parent_objs),fuzz-$(obj))
 
 # Create mconfig.h for fuzzing. because fuzz target not runned by top level make; we need to create mconfig.h manually.
-gen-mconfig:
-	$(MAKE) -C ../../../ mconfig
-	$(MAKE) -C ../../../build/tools/
-	../../../build/tools/mconfig-gen | tee includes/mconfig.h
+../../../mconfig:
+        $(MAKE) -C ../../../ mconfig
+
+../../../build/includes/mconfig.h: ../../../mconfig
+        $(MAKE) -C ../../../build all
+
+include/mconfig.h: ../../../build/includes/mconfig.h
+        /usr/bin/env mv ../../../build/includes/mconfig.h includes/mconfig.h
 
 fuzz: prepare-incdir gen-mconfig fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
 	clang++ -std=c++11 -g -O1 -Iincludes -I../../../dasynq/include -fsanitize=fuzzer,address,undefined,leak fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects) -o fuzz

--- a/src/tests/cptests/Makefile
+++ b/src/tests/cptests/Makefile
@@ -38,7 +38,13 @@ clean:
 fuzz_parent_test_objects = $(foreach obj,$(notdir $(parent_test_objects)),fuzz-$(obj))
 fuzz_objects = $(foreach obj,$(parent_objs),fuzz-$(obj))
 
-fuzz: prepare-incdir fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
+# Create mconfig.h for fuzzing. because fuzz target not runned by top level make; we need to create mconfig.h manually.
+gen-mconfig:
+	$(MAKE) -C ../../../ mconfig
+	$(MAKE) -C ../../../build/tools/
+	../../../build/tools/mconfig-gen | tee includes/mconfig.h
+
+fuzz: prepare-incdir gen-mconfig fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
 	clang++ -std=c++11 -g -O1 -Iincludes -I../../../dasynq/include -fsanitize=fuzzer,address,undefined,leak fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects) -o fuzz
 
 $(fuzz_parent_test_objects): fuzz-%.o: ../%.cc

--- a/src/tests/cptests/Makefile
+++ b/src/tests/cptests/Makefile
@@ -40,13 +40,13 @@ fuzz_objects = $(foreach obj,$(parent_objs),fuzz-$(obj))
 
 # Create mconfig.h for fuzzing. because fuzz target not runned by top level make; we need to create mconfig.h manually.
 ../../../mconfig:
-        $(MAKE) -C ../../../ mconfig
+	$(MAKE) -C ../../../ mconfig
 
 ../../../build/includes/mconfig.h: ../../../mconfig
-        $(MAKE) -C ../../../build all
+	$(MAKE) -C ../../../build all
 
 include/mconfig.h: ../../../build/includes/mconfig.h
-        /usr/bin/env mv ../../../build/includes/mconfig.h includes/mconfig.h
+	/usr/bin/env mv ../../../build/includes/mconfig.h includes/mconfig.h
 
 fuzz: prepare-incdir gen-mconfig fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects)
 	clang++ -std=c++11 -g -O1 -Iincludes -I../../../dasynq/include -fsanitize=fuzzer,address,undefined,leak fuzz.cc $(fuzz_parent_test_objects) $(fuzz_objects) -o fuzz

